authentication/http_recordings_test.go

```diff
@@ -41,6 +44,30 @@ func configureHTTPTestRecordings(t *testing.T) {
 
 	authAPI.http.Transport = recorderTransport
 
+	// Set a custom matcher that will ensure the request body matches the recording, as both of
```

Comment:
Could we use [this testify assert](https://pkg.go.dev/github.com/stretchr/testify/require?utm_source=godoc#JSONEq) to compare JSON (when it's JSON)?

Reply:
Ah nice that looks to handle the JSON structure being different, will change to use that if it smells like JSON

Reply:
Yeah I think we can improve the checking of the form values too but it might be a bit more awkward as the request will only have them in the encoded formdata format not a map. It might be possible to use [ParseForm](https://pkg.go.dev/net/http#Request.ParseForm)